src/documents/management/commands/document_correspondents.py

```diff
@@ -0,0 +1,45 @@
+from django.core.management.base import BaseCommand
+
+from documents.models import Document, Correspondent
+
+from ...mixins import Renderable
+
+
+class Command(Renderable, BaseCommand):
+    help = """
+        Using the current set of correspondent rules, apply said rules to all
+        documents in the database, effectively allowing you to back-tag all
+        previously indexed documents with correspondent created (or modified)
+        after their initial import.
+    """.replace("    ", "")
+
+    def __init__(self, *args, **kwargs):
+        self.verbosity = 0
+        BaseCommand.__init__(self, *args, **kwargs)
+
+    def handle(self, *args, **options):
+
+        self.verbosity = options["verbosity"]
+
+        for document in Document.objects.all():
+            # No matching correspondents, so no need to continue
+            if document.correspondent:
+                continue
+
+            potential_correspondents = list(
+                Correspondent.match_all(document.content))
+            if not potential_correspondents:
+                continue
+
+            potential_count = len(potential_correspondents)
+
+            selected = potential_correspondents[0]
```

Comment:
Won't it be a little annoying for the user to specify the right correspondent for every file ?

Maybe add an option to leave it empty by default if there are multiple matched correspondent ?